### DISCUSSION TITLE
Issue with $persist in a Store

### DISCUSSION THREAD

**Discussion by robdekort** [alpinejs/alpine#2317] [1. Help] [answered]
Hey folks. It could be I'm getting the syntax wrong here. I was playing around with v3.5 and tried the following code:

```js
document.addEventListener('alpine:init', () => {
    Alpine.store('cookieBanner', {
        open: Alpine.$persist(true).as('cookieBannerOpen'),
        consent: Alpine.$persist(false).as('cookieBannerConsent'),
        consentDate: Alpine.$persist(false).as('cookieBannerConsentDate'),
        analyticsStorage: Alpine.$persist(false).as('cookieBannerAnalyticsStorage'),
        adStorage: Alpine.$persist(false).as('cookieBannerAdStorage'),
    })
})
```

This resulted in only the last variable persisting (`adStorage`) and not the others. I wonder if this could be a bug,  a wrong way to use it or me getting it totally wrong.

**Comment by SimoTod** [comment, marked as answer]
Definitely a bug. I've raised a pr to fix it but in the meantime it works if you don't use aliases

**Reply by robdekort** [reply]
Thank you. I appreciate your help. Yeah I just noticed that a few minutes ago. However I also noticed with the above snippet and no aliases is that it set's _all_ properties to false. Even the `open` one which i explicitly set to `true`.

Thanks for the PR. You rock!
